internal/datacoord/compaction.go

```diff
@@ -165,6 +166,21 @@ func summaryCompactionState(tasks []*datapb.CompactionTask) *compactionInfo {
 	return ret
 }
 
+func (c *compactionPlanHandler) checkAndSetSegmentStating(segmentID int64) bool {
```

Comment:
When there are many executing tasks, the performance of this method is suboptimal. It is recommended to use a concurrentMap.